apex_launchtest/apex_launchtest/asserts/assert_output.py

```diff
@@ -126,22 +44,20 @@ def assertInStdout(proc_output,
     an error.
     :type strict_proc_matching: bool
     """
-    # Depending on the type of 'proc' we're going to dispatch this a little differently
-    if isinstance(proc, launch.actions.ExecuteProcess):
-        _assertInStdoutByProcessAction(
-            proc_output,
-            msg,
-            proc
-        )
-    elif isinstance(proc, str):
-        _assertInStdoutByStringProcessName(
-            proc_output,
-            msg,
-            proc,
-            cmd_args,
-            strict_proc_matching
-        )
+    resolved_procs = resolveProcesses(
+        info_obj=proc_output,
+        proc=proc,
+        cmd_args=cmd_args,
+        strict_proc_matching=strict_proc_matching
+    )
+
+    for proc in resolved_procs:  # Nominally just one matching proc
+        for output in proc_output[proc]:
+            if msg in output.text.decode('ascii'):
```

Comment:
To be more specific, why `'ascii'` is used, rather than passing no arguments (which is something we do in launch), which will get you `'utf-8'`:

https://docs.python.org/3/library/stdtypes.html#bytes.decode

Reply:
This is a leftover from hacking around with the process_io event.  You're correct that it should probably be "passing no arguments"  I'll open an issue to update this in a different PR